src/playground/PlaygroundLoader.tsx

```diff
@@ -301,6 +309,18 @@ function initState(
 		files = defaultPlaygroundState.files;
 	}
 
+	// handle rule domains
+	const ruleDomains: Record<string, string> = {};
+	for (const key of searchParams.keys()) {
+		if (key.startsWith("ruleDomains.")) {
+			const domain = key.slice("ruleDomains.".length);
```

Comment:
imho: you might consider moving `"ruleDomains.".length` outside the loop to avoid recalculating the slice length on every iteration.

```suggestion
        const prefixLength = "ruleDomains.".length;
	for (const key of searchParams.keys()) {
		if (key.startsWith("ruleDomains.")) {
			const domain = key.slice(prefixLength);
```